### PR TITLE
Feat(Foundation/Tooltip): Added cursor exit paths

### PR DESCRIPTION
**Pull request by RostiMelk**
Tooltips

- Added props for alignX and alignY
- Added a 200ms delay for when the tooltip should show, and a 200ms delay for exit. The exit delay is is for when the user wants to select the text inside the tooltip, without the tool tip disappearing. 
- Accessibility arias and roles.

### CODE REVIEW

**Comment by RostiMelk** [comment]
Will also be working on reversing the tooltip direction if the tooltip is out of viewport bounds in this PR.

File: src/stories/Foundation/Tooltip/Tooltip.stories.tsx [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
 const Wrapper = styled.div`
-  margin-top: var(--spacing-32);
+  display: flex;
+  margin-top: var(--spacing-48);
+  justify-content: center;
 `;
 
-const BasicTemplate = () => {
+const Template = ({ alignX, alignY }) => {
+  const [veryCount, setVeryCount] = useState(1);
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/Tooltip.stories.tsx:15]: love verycount <3

File: src/stories/Foundation/Tooltip/Tooltip.stories.tsx [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+
+  const getTooltipText = () => {
+    const veryStr = Array(veryCount).fill('very').join(' ');
+    return `A ${veryStr} cool tooltip! ✨`;
+  };
+
+  const addVery = () => {
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/Tooltip.stories.tsx:22]: love how you name this function add so it is very easy to see that you add one! 💡

File: src/stories/Foundation/Tooltip/alignments.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
+export type AlignX = 'left' | 'center' | 'right';
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/alignments.ts:1]: good use of types!

File: src/stories/Foundation/Tooltip/alignments.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  return `
+    ${vertical}: calc(100% + var(--gap));
+    left: 50%;
+    transform: translateX(-50%);
+  `;
+};
+
+export const alignMap: Record<AlignY, Record<AlignX, string>> = {
+  top: {
+    left: getTooltipCSS('bottom', 'left'),
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/alignments.ts:24]: this reuse of css is awesome!

File: src/stories/Foundation/Tooltip/alignments.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    center: getTooltipCenteredCSS('bottom'),
+    right: getTooltipCSS('bottom', 'right'),
+  },
+  bottom: {
+    left: getTooltipCSS('top', 'left'),
+    center: getTooltipCenteredCSS('top'),
+    right: getTooltipCSS('top', 'right'),
+  },
+  // Centered tooltip is a bit different than the others, so we need to handle it separately
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/alignments.ts:33]: good comment 👨‍🎨

File: src/stories/Foundation/Tooltip/alignments.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    left: `
+      left: 0;
+      top: 50%;
+      transform: translate(calc(-100% - var(--gap)), -50%);
+    `,
+    center: null,
+    right: `
+      right: 0;
+      top: 50%;
+      transform: translate(calc(100% + var(--gap)), -50%);
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/alignments.ts:45]: your math is still looking good 🤓

File: src/stories/Foundation/Tooltip/index.tsx [hunk 7]
```diff
@@ -1,60 +1,180 @@
...
+  width: 100%;
 `;
 
 export interface Props {
-  text: string;
   children: React.ReactNode;
   className?: string;
+  text: string;
+  /**
+   * @default 'center'
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/index.tsx:56]: nice with the default comments 👍

File: src/stories/Foundation/Tooltip/index.tsx [hunk 8]
```diff
@@ -1,60 +1,180 @@
...
-);
+export const Tooltip = ({
+  children,
+  className,
+  text,
+  alignX = 'center',
+  alignY = 'top',
+}: Props) => {
+  // alignX and alignY cannot be both 'center'
+  if (alignX === 'center' && alignY === 'center') {
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/index.tsx:76]: can never be to sure, good checking 🧑‍🚒

File: src/stories/Foundation/Tooltip/index.tsx [hunk 9]
```diff
@@ -1,60 +1,180 @@
...
+    alignY = 'top';
+    console.warn(
+      'Tooltip: alignX and alignY cannot be both "center". alignY has been set to "top".'
+    );
+  }
+
+  const tooltipId = useId();
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/index.tsx:83]: I like this new little hook you found 🐅

File: src/stories/Foundation/Tooltip/index.tsx [hunk 10]
```diff
@@ -1,60 +1,180 @@
...
+      wrapper.removeEventListener('touchend', reset);
+    };
+  }, [text, children, localAlignX, localAlignY]);
+
+  return (
+    <Wrapper className={className} role='tooltip'>
+      <ToolTipWrapper
+        alignX={localAlignX}
+        alignY={localAlignY}
+        aria-hidden={!isVisible}
```
Comment (amfestoy) [inline src/stories/Foundation/Tooltip/index.tsx:168]: going to miss your aria taggs 😢

**Review by amfestoy — APPROVED** [review]
Beautiful work I am ready to see you go 😢 

image

**Comment by RostiMelk** [comment]
> Beautiful work I am ready to see you go 😢
> 
> image

#### Good bye for now 😭
Goodbye gif

**Pull request merged**